packages/core/src/core.ts

```diff
@@ -284,3 +286,20 @@ export function saveState(name: string, value: any): void {
 export function getState(name: string): string {
   return process.env[`STATE_${name}`] || ''
 }
+
+export async function getIDToken(audience: string): Promise<string> {
+  try {
+    // New ID Token is requested from action service
+    let id_token_url: string = getIDTokenUrl()
+
+    debug(`ID token url is ${id_token_url}`)
+
+    let body: string = await postCall(id_token_url, audience)
+    let id_token = parseJson(body)
+    return id_token
+
+  } catch (error) {
+    setFailed(error.message)
```

Comment:
I wouldn't setFailed. I would just throw the exception and let the caller decide what to do